src/Razor/src/Microsoft.AspNetCore.Razor.Language/RazorProjectEngine.cs

```diff
@@ -59,17 +47,9 @@ public virtual RazorCodeDocument ProcessDesignTime(RazorProjectItem projectItem)
             return codeDocument;
         }
 
-        internal virtual RazorCodeDocument ProcessDesignTime(RazorSourceDocument source, IReadOnlyList<RazorSourceDocument> importSources, IReadOnlyList<TagHelperDescriptor> tagHelpers)
-        {
-            if (source == null)
-            {
-                throw new ArgumentNullException(nameof(source));
-            }
+        public abstract RazorParserOptions GetParserOptions(Action<RazorParserOptionsBuilder> builder = null);
```

Comment:
Ideally it would be nice if this worked like Roslyn's APIs i.e. options \ inputs can be statically computed without having to involve the parser \ compiler